Test Failure: ConcurrencyTestServlet.testWebDDManagedExecutorDefinition

Test Failure (20220125-2232): test.jakarta.concurrency.ConcurrencyTest.ConcurrencyTestServlet.testWebDDManagedExecutorDefinition

```
junit.framework.AssertionFailedError: 2022-01-26-04:58:52:216 ERROR: Caught exception attempting to call test method testWebDDManagedExecutorDefinition on servlet test.jakarta.concurrency.web.ConcurrencyTestServlet
javax.naming.NamingException: CWNEN1001E: The object referenced by the java:global/concurrent/dd/web/LPExecutor JNDI name could not be instantiated. If the reference name maps to a JNDI name in the deployment descriptor bindings for the application performing the JNDI lookup, make sure that the JNDI name mapping in the deployment descriptor binding is correct. If the JNDI name mapping is correct, make sure the target resource can be resolved with the specified name relative to the default initial context. [Root exception is com.ibm.wsspi.injectionengine.InjectionException: CWNEN0030E: The server was unable to obtain an object instance for the java:global/concurrent/dd/web/LPExecutor reference.  The exception message was: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp]
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.newCannotInstantiateObjectException(InjectionJavaColonHelper.java:183)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:118)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:332)
	at com.ibm.ws.jndi.url.contexts.javacolon.internal.JavaURLContext.lookup(JavaURLContext.java:370)
	at org.apache.aries.jndi.DelegateContext.lookup(DelegateContext.java:149)
	at java.naming/javax.naming.InitialContext.lookup(InitialContext.java:409)
	at java.naming/javax.naming.InitialContext.doLookup(InitialContext.java:282)
	at test.jakarta.concurrency.web.ConcurrencyTestServlet.testWebDDManagedExecutorDefinition(ConcurrencyTestServlet.java:2228)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at componenttest.app.FATServlet.doGet(FATServlet.java:71)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1258)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:746)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:443)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1227)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1011)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:85)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:938)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:281)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1184)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:453)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:412)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:566)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:500)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:360)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:327)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:245)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.ibm.wsspi.injectionengine.InjectionException: CWNEN0030E: The server was unable to obtain an object instance for the java:global/concurrent/dd/web/LPExecutor reference.  The exception message was: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1454)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1384)
	at com.ibm.ws.injectionengine.osgi.internal.naming.InjectionJavaColonHelper.getObjectInstance(InjectionJavaColonHelper.java:116)
Caused by: java.lang.Exception: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp
	at io.openliberty.concurrent.processor.AppDefinedResourceFactory.createResource(AppDefinedResourceFactory.java:94)
	at io.openliberty.concurrent.processor.AppDefinedResourceFactory.createResource(AppDefinedResourceFactory.java:118)
	at com.ibm.ws.injectionengine.osgi.internal.ResourceFactoryObjectFactory.getObjectInstance(ResourceFactoryObjectFactory.java:33)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObjectInstance(InjectionBinding.java:1552)
	at com.ibm.wsspi.injectionengine.InjectionBinding.getInjectionObject(InjectionBinding.java:1428)

	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:92)
	at componenttest.custom.junit.runner.SyntheticServletTest.invokeExplosively(SyntheticServletTest.java:42)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:199)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:320)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:173)
```

The managed executor definition is processed here,

```
[1/26/22, 4:58:29:249 PST] 00000020 id=00000000 y.concurrent.processor.ManagedExecutorResourceFactoryBuilder < createResourceFactory Exit  
                                                                                                               io.openliberty.concurrent.processor.AppDefinedResourceFactory@2a40ecac
```

and then the test application tries to use it here:

```
[1/26/22, 4:58:47:152 PST] 00000021 id=00000000 SystemOut                                                    O >>> BEGIN: testWebDDManagedExecutorDefinition
```

AppDefinedResourceFactory waits 5 seconds for it to appear in the service registry, but it is slow to be registered and so fails:

```
[1/26/22, 4:58:52:174 PST] 00000021 id=00000000 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.Exception: Unable to find managedExecutorService[java:global/concurrent/dd/web/LPExecutor] for application ConcurrencyTestApp io.openliberty.concurrent.processor.AppDefinedResourceFactory 96" at ffdc_22.01.26_04.58.52.0.log
```

A couple of seconds later, it does get registered and activated:

```
[1/26/22, 4:58:54:077 PST] 0000001b id=13c1b9d5 LogService-66-com.ibm.ws.concurrent.jakarta                  1 ServiceEvent REGISTERED 
...
[1/26/22, 4:58:54:078 PST] 0000001b id=4b8f172d com.ibm.ws.concurrent.internal.ManagedExecutorServiceImpl    < activate Exit
```

I'll look into increasing the amount of time that AppDefinedResourceFactory will wait for services to become available.

--- a/dev/io.openliberty.concurrent/src/io/openliberty/concurrent/processor/AppDefinedResourceFactory.java
+++ b/dev/io.openliberty.concurrent/src/io/openliberty/concurrent/processor/AppDefinedResourceFactory.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2013,2021 IBM Corporation and others.
+ * Copyright (c) 2013,2022 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -11,6 +11,7 @@
 package io.openliberty.concurrent.processor;
 
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.InvalidSyntaxException;
@@ -89,7 +90,7 @@ public class AppDefinedResourceFactory implements com.ibm.ws.resource.ResourceFa
         Object resource;
 
         try {
-            ResourceFactory factory = tracker.waitForService(5000);
+            ResourceFactory factory = tracker.waitForService(TimeUnit.MINUTES.toMillis(1));
             if (factory == null) // TODO this was MISSING_RESOURCE_J2CA8030 in com.ibm.ws.jca.cm messages
                 throw new Exception("Unable to find " + (info == null ? id : info.getType() + ' ' + id) + " for application " + appName);
             resource = factory.createResource(info);

